app/assets/v2/js/pages/profile-trust.js

```diff
@@ -139,3 +139,234 @@ let show_brightid_verify_modal = function(brightid_uuid) {
+          vm.verificationEnabled = false;
+          vm.verified = true;
+          vm.validationStep = 'verifyNumber';
+          vm.showValidation = false;
+          _alert('You have been verified. <a href="" style="color: white; text-decoration: underline;">Refresh Page</a>.', 'success');
+        }).catch((e) => {
+          vm.errorMessage = e.responseJSON.msg;
+        });
+      }
+    },
+    startVerification() {
+      this.phone = '';
+      this.validationStep = 'requestVerification';
+      this.validNumber = false;
+      this.errorMessage = '';
+      this.code = '';
+      this.timePassed = 0;
+      this.timeInterval = 0;
+      this.display_email_option = false;
+    },
+    countdown() {
+      const vm = this;
+
+      if (!vm.countDownActive) {
+        vm.countDownActive = true;
+
+        setInterval(() => {
+          vm.timePassed += 1;
+        }, 1000);
+      }
+    },
+    resendCode(delivery_method) {
+      const e164 = this.phone.replace(/\s/g, '');
+      const vm = this;
+
+      vm.errorMessage = '';
+
+      if (vm.validNumber) {
+        const verificationRequest = fetchData('/sms/request', 'POST', {
+          phone: e164,
+          delivery_method: delivery_method || 'sms'
+        }, {'X-CSRFToken': vm.csrf});
+
+        vm.errorMessage = '';
+
+        $.when(verificationRequest).then(response => {
+          // set the cooldown time to one minute
+          this.timePassed = 0;
+          this.timeInterval = 60;
+          this.countdown();
+          this.display_email_option = response.allow_email;
+        }).catch((e) => {
+          vm.errorMessage = e.responseJSON.msg;
+        });
+      }
+    },
+    // REQUEST VERIFICATION
+    requestVerification(event) {
+      const e164 = this.phone.replace(/\s/g, '');
+      const vm = this;
+
+      if (vm.validNumber) {
+        const verificationRequest = fetchData('/sms/request', 'POST', {
+          phone: e164
+        }, {'X-CSRFToken': vm.csrf});
+
+        vm.errorMessage = '';
+
+        $.when(verificationRequest).then(response => {
+          vm.validationStep = 'verifyNumber';
+          this.timePassed = 0;
+          this.timeInterval = 60;
+          this.countdown();
+          this.display_email_option = response.allow_email;
+        }).catch((e) => {
+          vm.errorMessage = e.responseJSON.msg;
+        });
+      }
+    },
+    isValidNumber(validation) {
+      console.log(validation);
+      this.validNumber = validation.isValid;
+    }
+  }
+});
+
+$(document).ready(function() {
+  $(document).on('keyup', 'input[name=telephone]', function(e) {
+    var number = $(this).val();
+
+    if (number[0] != '+') {
+      number = '+' + number;
+      $(this).val(number);
+    }
+  });
+
+
+  $(document).on('click', '#verify_offline', function(e) {
+    $(this).remove();
+    $('#verify_offline_target').css('display', 'block');
+  });
+});
+
+if (document.getElementById('gc-sms-modal')) {
+
+  const app = new Vue({
```

Comment:
can we do something like const appTrust  here? I want to avoid coalitions with others apps loaded on the same page.